src/complex/Parameters/FileSystemPathParameter.cpp

```diff
@@ -73,6 +73,10 @@ Result<> ValidateDirectoryWritePermission(const FileSystemPathParameter::ValueTy
   while(!fs::exists(checkedPath))
   {
     checkedPath = checkedPath.parent_path();
+    if(checkedPath.empty())
```

Comment:
@imikejackson I think we essentially solved it with the same checks.  I put the absolute vs relative path checks down in the main validation function because you might run in to relative paths you can't for input files/directories on top of output in which case the ValidateDirectoryWritePermission fct won't get hit.  Other than that though I like your updates better since your error messaging is more descriptive.